pennylane/templates/subroutines/qdrift.py

```diff
@@ -188,42 +199,13 @@ def __init__(  # pylint: disable=too-many-arguments
             "base": hamiltonian,
             "decomposition": decomposition,
         }
-        super().__init__(time, wires=hamiltonian.wires, id=id)
+        super().__init__(*hamiltonian.data, time, wires=hamiltonian.wires, id=id)
```

Comment:
This makes the `QDrift.data` consistent with `_flatten`. We need to make this change to a number of similar templates below.

Reply:
Mind double checking `qml.ops.functions.bind_new_parameters` works properly too?

After #5701  we can update `bind_new_parameters` to use the pytrees instead which will make `bind_new_parameters` much more robust. But I wouldn't hold my breath on that update.

Reply:
Sure, will do that! Should that maybe be a part of `assert_valid`? :thinking: (probably annoying to create new parameters for an arbitrary op)

Reply:
`bind_new_parameters is already called in `assert_valid`, but would be good to double check since the check in assert valid uses all 0s for the new parameters

Reply:
Yeah, the check does not consider `hyperparameters`, which the Hamiltonian of QDrift is one of.
I'll fix `bind_new_parameters` for `QDrift` in this PR but will not check other templates. This is left for future work and it's time to converge on this PR.